reaper/cluster.go

```diff
@@ -0,0 +1,243 @@
+package reaper
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"runtime"
+	"sync"
+	"time"
+)
+
+type Cluster struct {
+	Name           string
+	JmxUsername    string
+	JmxPasswordSet bool
+	Seeds          []string
+	NodeState      NodeState
+}
+
+type NodeState struct {
+	GossipStates []GossipState
+}
+
+type GossipState struct {
+	SourceNode    string
+	EndpointNames []string
+	TotalLoad     float64
+	DataCenters   map[string]DataCenterState
+}
+
+type DataCenterState struct {
+	Name  string
+	Racks map[string]RackState
+}
+
+type RackState struct {
+	Name      string
+	Endpoints []EndpointState
+}
+
+type EndpointState struct {
+	Endpoint       string
+	DataCenter     string
+	Rack           string
+	HostId         string
+	Status         string
+	Severity       float64
+	ReleaseVersion string
+	Tokens         string
+	Load           float64
+}
+
+type GetClusterResult struct {
+	Cluster *Cluster
+	Error   error
+}
+
+type RepairSchedule struct {
+	Id                string    `json:"id"`
+	Owner             string    `json:"owner,omitempty"`
+	State             string    `json:"state,omitempty"`
+	Intensity         float64   `json:"intensity,omitempty"`
+	ClusterName       string    `json:"cluster_name,omitempty"`
+	KeyspaceName      string    `json:"keyspace_name,omitempty"`
+	RepairParallism   string    `json:"repair_parallelism,omitempty"`
+	IncrementalRepair bool      `json:"incremental_repair,omitempty"`
+	ThreadCount       int       `json:"repair_thread_count,omitempty"`
```

Comment:
Is the Repair prefix needed here for the json property name?

Reply:
We're tied to the existing json format so there's not much we can change here without changing Reaper itself.

Reply:
But this raises the question: should the field be renamed to `RepairThreadCount`? Does the prefix add a valuable context here?

Reply:
If you don't add it, then you make a choice in diverging from the json field names. If you do it here, then it probably means you need to remove the `Repair` prefixes everywhere.
I'm happy either way as long as it's consistent.

Reply:
OK I will add the prefix. 

Note that the API design doc has these prefixes already. Here we are talking about the legacy Cluster API that has not yet been retrofitted to comply with the API principles.

Reply:
Done in f373d62.